File: 0066-plus-one/0066-plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function (digits) {
    let num = digits.reduce((acc, val) => {
        return acc + val;
    }, "");
    num = BigInt(num) + BigInt(1);
    console.log(num.toString().split(""));
    return num.toString().split("");
};

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length -1 ;
        for ( int i = n  ; i >=0 ; i--){
            int sum = digits[i] +1;

            if ( sum <= 9){
                digits[i]++ ;
                return digits;
            }
            digits[i]=0;
        }
        int res []= new int [n+2];
        res[0] = 1;
        return res;
    }
}